crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -726,6 +728,20 @@ impl<'this, TX: DbTx<'this>> AccountExtReader for DatabaseProvider<'this, TX> {
     }
 }
 
+impl<'this, TX: DbTx<'this>> AccountChangeReader for DatabaseProvider<'this, TX> {
+    fn account_block_changeset(&self, block_number: BlockNumber) -> Result<Vec<AccountBeforeTx>> {
+        let range = block_number..=block_number;
+        self.tx
+            .cursor_read::<tables::AccountChangeSet>()?
+            .walk_range(range)?
+            .map(|result| -> Result<_> {
+                let (_, account_before) = result?;
+                Ok(account_before)
+            })
```

Comment:
This could be:

```suggestion
            .map(|result| -> Result<_> { Ok(result?.1) })
```

Or even a nested map on the result but personally I find the separate destructuring line more readable.

Reply:
Looks good!